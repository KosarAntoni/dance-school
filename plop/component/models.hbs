export type {{properCase name}}Props {

}