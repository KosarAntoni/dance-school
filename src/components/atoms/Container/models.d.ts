import { ElementType, RefObject } from 'react';

export type ContainerProps = {
  as?: ElementType;
  className?: string;
  outerClassName?: string;
  gutter?: boolean;
  variant?: 'wide' | 'fluid' | string;
  innerRef?: RefObject;
};
