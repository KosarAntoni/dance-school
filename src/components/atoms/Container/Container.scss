@import 'styles/component-set';

.container {
  max-width: $wrapper-default;

  &__outer {
    display: flex;
    justify-content: center;
    width: 100%;
  }

  &--fluid {
    &.container {
      &__inner {
        max-width: 100%;
      }
    }
  }

  &--wide {
    &.container {
      &__inner {
        max-width: $wrapper-wide;
      }
    }
  }

  &--gutter {
    padding-right: $gap-default;
    padding-left: $gap-default;
  }
}
