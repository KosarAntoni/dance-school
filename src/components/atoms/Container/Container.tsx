import React, { FC } from 'react';
import classnames from 'classnames';

import { ContainerProps } from './models';

import './Container.scss';

const Container: FC<ContainerProps> = ({
  as,
  className,
  outerClassName,
  variant,
  gutter = true,
  innerRef,
  children,
  ...rest
}) => {
  const Component = as || 'div';
  const containerOuterClassNames = classnames(
    'container__outer',
    {
      [`container--${variant}`]: variant,
      'container--gutter': gutter,
    },
    outerClassName
  );
  const containerClassNames = classnames('container', className);

  return (
    <Component className={containerOuterClassNames} data-testid="container-item">
      <div className={containerClassNames} ref={innerRef} {...rest}>
        {children}
      </div>
    </Component>
  );
};

export default Container;
