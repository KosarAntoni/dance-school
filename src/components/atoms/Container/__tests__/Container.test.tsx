import React from 'react';
import { render, screen } from '@testing-library/react';

import {
  ContainerChildrenMock,
  ContainerDefaultClassesMock,
  ContainerDefaultPropsMock,
  ContainerOuterDefaultClassesMock,
} from '../__mock__/mock';
import Container from '../Container';

import { ContainerProps } from '../models';

const defaultProps: ContainerProps = {
  ...ContainerDefaultPropsMock,
};

test('the Container component renders correctly with required and optional props', () => {
  render(<Container {...defaultProps}>{ContainerChildrenMock}</Container>);

  const containerNode = screen.getByTestId('container-item');

  // the component has been rendered
  expect(containerNode).toBeInTheDocument();

  // generating a snapshot, not for testing purposes but only to see what changes in the HTML structure during CR if the developer updates the snapshot
  expect(containerNode).toMatchSnapshot();

  // the comonent has correct classes
  expect(containerNode).toHaveClass(ContainerOuterDefaultClassesMock.join(' '));
  expect(containerNode.firstChild).toHaveClass(ContainerDefaultClassesMock.join(' '));

  // the component contains text
  const childNode = screen.getByText(ContainerChildrenMock);
  expect(childNode).toBeInTheDocument();
});
