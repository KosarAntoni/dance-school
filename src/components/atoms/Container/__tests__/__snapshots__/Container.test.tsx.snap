// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Container component renders correctly with required and optional props 1`] = `
<div
  class="container__outer container--wide container--gutter outer-test-class"
  data-testid="container-item"
>
  <div
    class="container test-class"
  >
    test content
  </div>
</div>
`;
